pkg/helm/cache.go

```diff
@@ -163,23 +156,6 @@ func helmAppFromSecret(secret *corev1.Secret) (*apptypes.HelmApp, error) {
 	_, helmApp.IsConfigurable = configSecret.Data["config"] // TODO: also check if there are any config items
 	helmApp.ChartPath = string(configSecret.Data["chartPath"])
 
-	if configSecret.Data["license"] == nil {
```

Comment:
`helmAppFromSecret` no longer mutates anything.  License is download is now triggered explicitly when license is needed.